webapp/src/ts/modules/contacts/contacts-content.component.ts

```diff
@@ -153,4 +199,182 @@ export class ContactsContentComponent implements OnInit, OnDestroy {
     this.tasksTimeWindowWeeks = weeks;
     this.taskEndDate = weeks ? moment().add(weeks, 'weeks').format('YYYY-MM-DD') : null;
   }
+
+  private async setRightActionBar() {
```

Comment:
I'd prefer it if we don't mix then/catch chains and async/await in the same file. Can you please switch this to be promise based? 
Thanks!

Reply:
Actually prefer this one. it's working fine, readable, keeping code flat. We're going to start moving to async/await so files will have mix. :)